src/github.com/couchbase/sync_gateway/db/change_listener.go

```diff
@@ -85,6 +87,24 @@ func (listener *changeListener) Notify(keys base.Set) {
 	}
 	listener.tapNotifier.L.Lock()
 	listener.counter++
+	listener.terminateCheck = false
+	for key, _ := range keys {
+		listener.keyCounts[key] = listener.counter
+	}
+	base.LogTo("Changes+", "Notifying that %q changed (keys=%q) count=%d",
+		listener.bucket.GetName(), keys, listener.counter)
+	listener.tapNotifier.Broadcast()
+	listener.tapNotifier.L.Unlock()
+}
+
+// Changes the counter, notifying waiting clients.
+func (listener *changeListener) NotifyCheckForTermination(keys base.Set) {
+	if len(keys) == 0 {
+		return
+	}
+	listener.tapNotifier.L.Lock()
+	listener.counter++
```

Comment:
Do we need to update the listener counter here?  I had thought we specifically did not want to do that.